File: src/main/java/com/pygostylia/osprey/ChunkLocation.java
package com.pygostylia.osprey;

public record ChunkLocation(int x, int z) {
    public Double distanceFrom(ChunkLocation location) {
        return Math.sqrt(
            Math.pow(Math.abs(location.x - x), 2) +
            Math.pow(Math.abs(location.z - z), 2)
        );
    }
}

File: src/main/scala/com/pygostylia/osprey/ChunkLocation.scala
package com.pygostylia.osprey

final case class ChunkLocation(x: Int, z: Int) {
  def distanceFrom(location: ChunkLocation): Double = {
    Math.sqrt(Math.pow(Math.abs(location.x - x), 2) + Math.pow(Math.abs(location.z - z), 2))
  }
}
